examples/ota-requestor-app/linux/main.cpp

```diff
@@ -254,18 +254,21 @@ bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier,
 
 void SendQueryImageCommand(chip::NodeId peerNodeId = providerNodeId, chip::FabricIndex peerFabricIndex = providerFabricIndex)
 {
-    Server * server                        = &(Server::GetInstance());
+    Server * server           = &(Server::GetInstance());
+    chip::FabricInfo * fabric = server->GetFabricTable().FindFabricWithIndex(peerFabricIndex);
+
     chip::DeviceProxyInitParams initParams = {
         .sessionManager = &(server->GetSecureSessionManager()),
         .exchangeMgr    = &(server->GetExchangeManager()),
         .idAllocator    = &(server->GetSessionIDAllocator()),
-        .fabricInfo     = server->GetFabricTable().FindFabricWithIndex(providerFabricIndex),
+        .fabricInfo     = fabric,
         // TODO: Determine where this should be instantiated
         .imDelegate = chip::Platform::New<chip::Controller::DeviceControllerInteractionModelDelegate>(),
     };
 
-    chip::OperationalDeviceProxy * operationalDeviceProxy =
-        new chip::OperationalDeviceProxy(initParams, PeerId().SetNodeId(providerNodeId));
+    PeerId peerID = fabric->GetPeerId();
+    peerID.SetNodeId(peerNodeId);
+    chip::OperationalDeviceProxy * operationalDeviceProxy = new chip::OperationalDeviceProxy(initParams, peerID);
```

Comment:
need a null check?

Reply:
Also, this may not cut'n'paste well to embedded platforms where direct `new` usage is not done. Hopefully there will not be too much of that.

Reply:
Yes, this should be changed to `Platform::New`. The original code had `new`. I can update it to call `Platform::New` in this PR, with a null check.